### PR TITLE
Increase Dashboard memory limits

### LINKED ISSUE #1431: Errors for large numbers of objects
I've seen reports of the dashboard throwing errors and otherwise not working for large numbers of objects. 

Anecdotally, I have seen reports for large numbers of pods and large numbers of events. It may behove us to test with large numbers of objects and see which are problematic and start creating issues and fixing them.

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**: Increases memory requests and limits for Dashboard.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes https://github.com/kubernetes/dashboard/issues/1431

**Special notes for your reviewer**: Dashboard crashes on large clusters, this change should fix that problem.

**Release note**:

```release-note
Increase Dashboard's memory requests and limits
```

### CODE REVIEW

**Comment by maciaszczykm** [comment]
CC @kubernetes/dashboard-maintainers

File: cluster/addons/dashboard/dashboard-controller.yaml [hunk 1]
```diff
@@ -25,10 +25,10 @@ spec:
           # keep request = limit to keep this container in guaranteed class
```
Comment (pdecat) [inline cluster/addons/dashboard/dashboard-controller.yaml]: Now that memory request and limit are different, that comment is probably wrong.

Comment (maciaszczykm) [inline cluster/addons/dashboard/dashboard-controller.yaml]: Yes, you are right. Should I fix it or you will take care of it?

Comment (pdecat) [inline cluster/addons/dashboard/dashboard-controller.yaml]: Well, it depends on what is desired: either we fix the comment or the limits to restore the truth.

I believe the latter is the best option to get the Guaranteed QoS on that pod:

> If limits and optionally requests (not equal to 0) are set for all resources across all containers and they are equal, then the pod is classified as Guaranteed.

https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-qos.md#qos-classes

**Pull request merged**